ming/mim.py

```diff
@@ -479,17 +495,8 @@ def insert_many(self, documents, ordered=True, session=None):
         result = self.__insert(documents)
         return InsertManyResult(result, True)
 
-    def save(self, doc, **kwargs):
-        warnings.warn('save is now deprecated, please use insert_one or replace_one', DeprecationWarning, stacklevel=2)
-        _id = doc.get('_id', ())
-        if _id == ():
-            return self.__insert(doc)
-        else:
-            self.__update({'_id':_id}, doc, upsert=True)
-            return _id
-
     def replace_one(self, filter, replacement, upsert=False):
-        return self._update(filter, replacement, upsert)
+        return self.__update(filter, replacement, upsert)
 
     def __update(self, spec, updates, upsert=False, multi=False):
```

Comment:
`def __update` returns a dict with some values like `err` `n` `nModified` etc.  That is old.  Now pymongo returns a UpdateResult https://pymongo.readthedocs.io/en/stable/api/pymongo/results.html#pymongo.results.UpdateResult

MIM should be updated to match

Ming _could_ be an abstraction layer and return the old format, but I think there's probably not _that_ many places that use the old result dict, and we could require them to be updated.

Reply:
I modified all of these internal mim methods to natively pass around pymongo Update/Delete/InsertResult instances. 

The `raw_result` dict that both `UpdateResult` and `DeleteResult` use keeps these old field names like: `'n'` and `'nModified'`. So I brought those original dict shapes back internally in mim. Then mim passes those `raw_result` dicts into their corresponding `Result` class.